core/src/main/java/com/linecorp/armeria/internal/common/metric/RequestMetricSupport.java

```diff
@@ -373,6 +397,23 @@ public DistributionSummary failureAttempts() {
                                                             idPrefix.name("actual.requests.attempts"),
                                                             idPrefix.tags("result", "failure"));
         }
+
+        @Override
+        public DistributionSummary failureAttempts(Throwable error) {
+            final String causeName = error.getClass().getSimpleName();
+
+            if (failureAttemptsWithErrors.containsKey(causeName)) {
+                return failureAttemptsWithErrors.get(causeName);
+            }
+
+            final DistributionSummary distributionSummary =
+                    newDistributionSummary(parent,
+                                           idPrefix.name("actual.requests.attempts"),
+                                           idPrefix.tags("result", "failure",
+                                                         "cause", causeName));
```

Comment:
Should we use `Counter` like we did in `responseTimeouts()` and `writeTimeouts()`?

Reply:
I considered using `Counter`. But I refered to the implementation of `failureAttempts()` and implemeted like this PR.
In this case, should we use `Counter` when counting failure attempts with errors and record failure attempts whether the error is null or not by using `failureAttempts()`?

Reply:
I think the purpose of the two metrics is different.
- `failureAttemps()` tells how many attempts have been made before a request fails.
- The counter of the causes of failure attempts provides the trend of exception for each retry. That said, the current implementation recodes the last `log.responseCause()` then a request fails.

I imagined:
```
for (var child : log.children()) {
   // Increase a counter for the exception
   child.responseCause()
   // If responseCause is null, the response status could be a fallback value.
}
```